File: src/main/resources/generators/scala/native/StringWrapper.scala
class StringWrapper(private val inner: String) {
  def nonEmpty(): BooleanWrapper = inner.nonEmpty

  def trim(): StringWrapper = new StringWrapper(inner.trim)

  def startsWith(prefix: StringWrapper): BooleanWrapper = inner.startsWith(prefix.inner)

  def matches(regex: StringWrapper): BooleanWrapper = inner.matches(regex.inner)
}

File: src/main/resources/generators/ts/native/StringWrapper.ts
export class StringWrapper{
    private inner: string;
    
    constructor(inner: string) {
        this.inner = inner || "";
    }

    toString() {
        return this.inner;
    }
    
    nonEmpty(): boolean {
        return this.inner.length > 0;
    }

    trim(): StringWrapper {
        return new StringWrapper(this.inner.trim());
    }

    startsWith(prefix: StringWrapper): boolean {
        return this.inner.substr(0, prefix.inner.length) === prefix.inner;
    }

    matches(regex: StringWrapper): boolean {
        return new RegExp(regex.inner).test(this.inner);
    }
}